corehq/apps/users/tests/test_views.py

```diff
@@ -174,6 +176,78 @@ def test_cannot_update_parent_data(self):
             _update_role_from_view('test-domain', role_data)
 
 
+class DeleteRoleTests(TestCase):
+    def test_delete_role_removes_role(self):
+        role = UserRole.objects.create(domain=self.domain, name='test-role')
+
+        request = self._create_request({'_id': role.get_id})
+
+        delete_user_role(request, 'test-domain')
+
+        with self.assertRaises(UserRole.DoesNotExist):
+            UserRole.objects.get(couch_id=role.get_id)
+
+    # NOTE: Documenting current behavior. The API should probably be changed to return an error
+    def test_missing_role_returns_nothing(self):
+        request = self._create_request({'_id': 'invalid_id'})
+
+        response = delete_user_role(request, 'test-domain')
+        self.assertEqual(response.content.decode('utf-8'), '{}')
+
+    def test_cannot_delete_linked_domain_data(self):
+        role = UserRole.objects.create(domain='test-domain', name='test-role', upstream_id='parent_id')
+
+        request = self._create_request({'_id': role.get_id})
+
+        response = delete_user_role(request, 'test-domain')
+        remaining_role = UserRole.objects.get(id=role.id)
+        # Following existing behavior. This should return an error when the missing role also returns an error
+        self.assertEqual(response.content.decode('utf-8'), '{}')
```

Comment:
I know you are following existing behavior, but seeing as this part is new code you added in this PR, why not address the response as part of this PR?

Reply:
Most likely will be doing so